sample/trace_key.go

```diff
@@ -47,42 +51,34 @@ func newTraceKey(fields []string, useTraceLength bool) *traceKey {
 // returns the number of values used to build the key
 func (d *traceKey) build(trace *types.Trace) (string, int) {
 	fieldCount := 0
-	// fieldCollector gets all values from the fields listed in the config, even
-	// if they happen multiple times.
-	fieldCollector := make(map[string][]string)
 
 	// for each field, for each span, get the value of that field
 	spans := trace.GetSpans()
-	uniques := generics.NewSetWithCapacity[string](maxKeyLength)
+	uniques := newDistinctValue(d.fields, maxKeyLength)
 outer:
-	for _, field := range d.fields {
+	for i, field := range d.fields {
 		for _, span := range spans {
 			if val, ok := span.Data[field]; ok {
-				u := fmt.Sprintf("%s/%v", field, val)
 				// don't bother to add it if we've already seen it
-				if uniques.Contains(u) {
-					continue
-				}
-				uniques.Add(u)
-				if len(uniques) >= maxKeyLength {
+				if uniques.totalUniqueCount >= maxKeyLength {
 					break outer
 				}
-				fieldCollector[field] = append(fieldCollector[field], fmt.Sprintf("%v", val))
+				if uniques.AddAsString(val, i) {
+					continue
+				}
 			}
 		}
 	}
 	// ok, now we have a map of fields to a list of all unique values for that field.
 	// (unless it was huge, in which case we have a bunch of them)
 
 	var key strings.Builder
-	for _, field := range d.fields {
+	for i := range d.fields {
+		values := uniques.Values(i)
 		// if there's no values for this field, skip it
-		values, ok := fieldCollector[field]
-		if !ok || len(values) == 0 {
+		if len(values) == 0 {
 			continue
 		}
-		// sort and collapse list
-		sort.Strings(values)
```

Comment:
sorting was important before to ensure the list of values was always the same when possible. Does the new algorithm not need that assurance anymore?